### PR TITLE
Python 3.7 and 3.9 related fixes

### PR DESCRIPTION
**Pull request by richlowe**
These are the changes to fix things under Python 3.7 and 3.9 from @citrus-it and a fix from Pete Dennis at Oracle for a test failure which taking andy's changes verbatim would otherwise cause.  We don't switch to using, or otherwise deliver, 3.7 or 3.9 bits, we just take the fixes.

These (after Pete's change), pass the test suite, etc, when run under 3.5.

(looking at the commits, there's also a fix for mkcert to use Makefile.com that stuck in here, and out of the last pull request.)

As always, please keep this as separate commits so that syncing up with omnios, oracle, etc, isn't even more complicated than it is already.

And also as always, please @citrus-it let me know if I'm screwing anything up under your name here.

### CODE REVIEW

File: src/util/mkcert/Makefile [hunk 1]
```diff
@@ -22,15 +22,16 @@
 # Copyright (c) 2016, Oracle and/or its affiliates. All rights reserved.
 #
 
+include ../../Makefile.com
```
Comment (citrus-it) [inline src/util/mkcert/Makefile:25]: It's probably more to do with the compiler version in omnios or the warnings that we enable, but I also had to do:

https://github.com/omniosorg/pkg5/pull/391/commits/bdd13600ce1d50e9b82ff26857db8ee8e50d245c

**Review by citrus-it — COMMENTED** [review]
Brought back some memories, particularly the ABC stuff!

**Review by Toasterson — APPROVED** [review]
LGTM also good catch about the is vs. != operator. Those are clear bugs https://realpython.com/courses/python-is-identity-vs-equality/ Once @citrus-it approves, I'll merge

File: src/util/mkcert/Makefile (hunk 1 above)
Comment (richlowe) [reply src/util/mkcert/Makefile:25]: I think you must enable `-Wextra` too, and I didn't.

**Review by citrus-it — APPROVED** [review]

**Pull request merged**